### DISCUSSION TITLE
Need help with adding JS tag (canonical tag) in nextjs

### DISCUSSION THREAD

**Discussion by sarfarazsiddiquii** [vercel/next.js#69846] [Help]
### Summary

I want to add a canonical tag in my next js application but I'm stuck on this error,

`67791:1 Failed to execute 'write' on 'Document': It isn't possible to write into a document from an asynchronously-loaded external script unless it is explicitly opened. (anonymous)@67791:1`

I have tried various approaches.

1.
```
<Script
        src="//tags.orquideassp.com/--"
        strategy="--"
      />
```

2.
```
useEffect(() => {
    
    const script = document.createElement('script');
    script.src = "//tags.orquideassp.com/---";
    script.async = false; 
    script.defer = false; 
    document.head.appendChild(script);
    return () => {
      document.head.removeChild(script);
    };
  }, []);
```

3.
```
<Head>
</Head>
<script src="//tags.orquideassp.com/--" />
```

**Comment by romhenri** [comment]
This error is related to the moment of execution.

Try using "strategy=" in Script to set "afterInteractive" or "lazyOnload".

Like Docs: https://nextjs.org/docs/pages/api-reference/components/script

**Reply by sarfarazsiddiquii** [reply]
Yes I've tried that too, in the first try.
I'm getting same error

**Comment by icyJoseph** [comment]
Hi,

Is your script using, `document.write`?  That API is not very friendly to async/defer scripts, because these run after the DOM is complete. I think, if you are in control of that script, try to change it to use DOM manipulations, rather than `document.write`, or perhaps, try with strategy, `beforeInteractive`, not guaranteed that'd work though.
